aws-api-appsync/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLRequest.java

```diff
@@ -60,97 +51,34 @@
      * Constructor for AppSyncGraphQLRequest.
      * @throw AmplifyException if a ModelSchema can't be derived from the provided model class
      */
-    private AppSyncGraphQLRequest(Builder builder) throws AmplifyException {
+    private AppSyncGraphQLRequest(Builder builder) {
         super(builder.responseType, new GsonVariablesSerializer());
-        this.modelSchema = ModelSchema.fromModelClass(builder.modelClass);
-        this.selectionSet = SelectionSet.Factory.fromModelClass(builder.modelClass,
-                builder.operationType,
-                DEFAULT_DEPTH);
-        this.operationType = builder.operationType;
+        this.modelSchema = builder.modelSchema;
+        this.operation = builder.operation;
+        this.selectionSet = builder.selectionSet;
         this.variables = builder.variables;
         this.variableTypes = builder.variableTypes;
```

Comment:
Ought these to be `Immutable.of(build.variables)` and `Immutable.of(builder.variableTypes)`?